pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -1041,7 +1041,16 @@ func (configgen *ConfigGeneratorImpl) buildSidecarOutboundHTTPListenerOptsForPor
 	if pluginParams.Port.Port == 0 {
 		rdsName = listenerOpts.bind // use the UDS as a rds name
 	} else {
-		rdsName = fmt.Sprintf("%d", pluginParams.Port.Port)
+		if pluginParams.ListenerProtocol == plugin.ListenerProtocolAuto &&
+			util.IsProtocolSniffingEnabledForNode(node) && listenerOpts.bind != actualWildcard {
```

Comment:
@rshriram add a check here. PTAL

Reply:
how can . protocol be Auto and sniffing not be enabled for a node?